controllers/dspipeline_controller.go

```diff
@@ -576,7 +577,7 @@ func (r *DSPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			})).
 		// TODO: Add watcher for ui cluster rbac since it has no owner
 		WithOptions(controller.Options{
-			MaxConcurrentReconciles: config.DefaultMaxConcurrentReconciles,
+			MaxConcurrentReconciles: r.MaxConcurrentReconcilesParam,
```

Comment:
We still want the default behavior if it is not specified, i.e. the change should be backwards compatible, convert this into a function similar to: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/config/defaults.go#L113